### PR TITLE
Remove `VariationalForm` and `InitialState`

### LINKED ISSUE #6063: Remove legacy `VariationalForm`
### What is the expected enhancement?

When the `algorithms` module from Qiskit Aqua got migrated to Terra, the `VariationalForm` was included (possibly by accident).
Before the next release, this class will no longer be needed because all variational forms will have been refactored into `BlueprintCircuit`s (see also https://github.com/Qiskit/qiskit-nature/pull/66 for an example).

Thus, it would be good to remove the `VariationalForm` class.

Furthermore, the `preferred_init_points` argument, which is checked for in [`VariationalAlgorithm`](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/algorithms/variational_algorithm.py#L204) and set to a dummy value in [`NLocal`](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/circuit/library/n_local/n_local.py#L509) will no longer be needed.

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

Closes #6063 

### Details and comments

Removes some legacy Aqua code. Namely:
- the `VariationalForm` class
- ~the `preferred_init_point` attribute which used to be checked for in the `var_form` of a `VariationalAlgorithm`~ EDIT: actually we still need this. However, a refactoring of how this is implemented should be addressed in the future.
- I also remove the legacy `InitialState` object

### CODE REVIEW

File: releasenotes/notes/remove-variational-form-3ff4901bdde9ab1e.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+deprecations:
+  - |
+    The legacy class `VariationalForm` and option `preferred_init_point` were
+    removed.
```
Comment (woodsp-ibm) [inline releasenotes/notes/remove-variational-form-3ff4901bdde9ab1e.yaml:5]: A release not is not needed here - the logic coming from Aqua, however it got refactored before this first release out of Terra, will be addressed in a migration guide

File: qiskit/algorithms/minimum_eigen_solvers/vqe.py [hunk 2]
```diff
@@ -139,10 +138,6 @@ def __init__(self,
         if optimizer is None:
             optimizer = SLSQP()
 
-        # set the initial point to the preferred parameters of the variational form
-        if initial_point is None and hasattr(var_form, 'preferred_init_points'):
-            initial_point = var_form.preferred_init_points
-
```
Comment (woodsp-ibm) [inline qiskit/algorithms/minimum_eigen_solvers/vqe.py:145]: This logic needs to be retained for VQE to correctly start at the desired state as indicated from the var form and initial state combination. Pehaps we can find some better way going forwards but this is the mechanism we have now.

Comment (mrossinek) [reply qiskit/algorithms/minimum_eigen_solvers/vqe.py:145]: Which initial state and variational form will be using this? The new `UCC` class will not have a `preferred_init_points` attribute.

File: qiskit/circuit/library/n_local/n_local.py [hunk 3]
```diff
@@ -150,16 +149,9 @@ def __init__(self,
...
-                    raise TypeError('initial_state must be of type InitialState, but is '
-                                    '{}.'.format(type(initial_state)))
-            except ImportError as ex:
-                raise ImportError(
-                    'Could not import the qiskit.aqua.components.initial_states.'
-                    'InitialState. To use this feature Qiskit Aqua must be installed.'
-                ) from ex
+            if not isinstance(initial_state, QuantumCircuit):
+                raise TypeError('initial_state must be of type QuantumCircuit, but is '
+                                '{}.'.format(type(initial_state)))
```
Comment (Cryoris) [inline qiskit/circuit/library/n_local/n_local.py:154]: Can you remove this check? Per type hint it must be a circuit already 🙂

File: qiskit/circuit/library/n_local/n_local.py [hunk 4]
```diff
@@ -110,7 +109,7 @@ def __init__(self,
         Raises:
             ImportError: If an ``initial_state`` is specified but Qiskit Aqua is not installed.
             TypeError: If an ``initial_state`` is specified but not of the correct type,
-                ``qiskit.aqua.components.initial_states.InitialState``.
+                ``QuantumCircuit``.
```
Comment (Cryoris) [inline qiskit/circuit/library/n_local/n_local.py:112]: (and then also this)

File: qiskit/circuit/library/n_local/n_local.py [hunk 5]
```diff
@@ -98,10 +98,9 @@ def __init__(self,
             skip_final_rotation_layer: Whether a final rotation layer is added to the circuit.
             skip_unentangled_qubits: If ``True``, the rotation gates act only on qubits that
                 are entangled. If ``False``, the rotation gates act on all qubits.
-            initial_state: A `qiskit.aqua.components.initial_states.InitialState` object which can
-                be used to describe an initial state prepended to the NLocal circuit. This
-                is primarily for compatibility with algorithms in Qiskit Aqua, which leverage
-                this object to prepare input states.
+            initial_state: A `QuantumCircuit` object which can be used to describe an initial state
+                prepended to the NLocal circuit. This is primarily for compatibility with algorithms
+                in Qiskit's application modules, which leverage this object to prepare input states.
```
Comment (Cryoris) [inline qiskit/circuit/library/n_local/n_local.py:103]:
```suggestion
                prepended to the NLocal circuit.
```

**Review by Cryoris — APPROVED** [review]

**Pull request merged**